plugin/core/views.py

```diff
@@ -172,3 +178,42 @@ def text_document_range_formatting(view: sublime.View, region: sublime.Region) -
         "options": formatting_options(view.settings()),
         "range": region_to_range(view, region).to_lsp()
     })
+
+
+def minihtml(view: sublime.View, content: Union[str, dict], popup_width: Optional[int] = None) -> str:
+    """ Content can be a string or MarkupContent. `popup_width` will wrap text based on the provided width. """
+    if isinstance(content, str):
+        return text2html(view, content, popup_width)
+    elif isinstance(content, dict):
+        value = content.get("value") or ""
+        kind = content.get("kind")
+        if kind == "markdown":
+            return mdpopups.md2html(view, value)
+        else:
+            # must be plaintext
+            return text2html(view, value, popup_width)
+    else:
+        return ''
+
+
+def text2html(view: sublime.View, content: str, popup_width: Optional[int] = None) -> str:
+    content = html.escape(content).replace('\n', '<br>').replace('\t', '&nbsp;&nbsp;&nbsp;&nbsp;')
+    if popup_width is not None:
+        # if popup width is provided, calucalte how many caracters can fit on one line
+        wrap_width = int(popup_width / view.em_width())
+        content = "<br>".join(textwrap.wrap(content, width=wrap_width,
```

Comment:
The newlines and tabs are already replaced in `content`. This will most likely influence what `textwrap.wrap` is doing?
Shouldn't we apply `textwrap.wrap` at the very beginning of this function instead?

Reply:
But if we apply this at the beginning of the function, before `html.escape`, then it might screw up `html.escape`.
I have my doubts about whether this `textwrap.wrap` function always does the right thing...

Reply:
I can apply it at the beginning before the encoding. 
But **then** instead of `"<br>".join(textwrap.wrap...` I should use **"\n"**`.join(textwrap.wrap...` 
than in the next line the \n will be replaced by a <br> 
```
content = "\n".join(textwrap.wrap...
content = html.escape(content).replace('\n', '<br>')...
```

I inserted <br> tags after the encoding, because I don't wont to insert new lines, just so I could replace them with <br> in the next line :)

> This will most likely influence what textwrap.wrap is doing?

Can you be more specific on what you mean by *influence* what the text wrap is doing?
influence in what way? Are concerned about whitespace? What is your concern? :)

Reply:
Well, if we replace one space character with six &nbsp; characters, the line has five more characters from the POV of textwrap.wrap

Reply:
And because \n is replaced by <br>, the function does not encounter new lines.